elements/pl-matrix-component-input/pl-matrix-component-input.py

```diff
@@ -0,0 +1,426 @@
+                else:
+                    html_params['incorrect'] = True
+            except Exception:
+                raise ValueError('invalid score' + score)
+
+        with open('pl-matrix-component-input.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+
+    elif data['panel'] == 'submission':
+
+        parse_error = data['format_errors'].get(name, None)
+        html_params = {
+            'submission': True,
+            'label': label,
+            'parse_error': parse_error,
+            'uuid': pl.get_uuid()
+        }
+
+        # Create error message for each matrix entry
+        # when at least one of them has invalid format
+        if parse_error is not None:
+            sub_output = ''
+            a_tru = pl.from_json(data['correct_answers'].get(name, None))
+            m, n = np.shape(a_tru)
+            for i in range(m):
+                for j in range(n):
+                    each_entry_name = name + str(n * i + j + 1)
+                    a_sub = data['raw_submitted_answers'].get(each_entry_name, None)
+                    sub_output += '<p> entry (' + str(i + 1) + ', ' + str(j + 1) + ') = "' + str(a_sub) + '" '
+                    entry_parse_error = data['format_errors'].get(each_entry_name, None)
+                    if entry_parse_error is not None:
+                        sub_output += entry_parse_error
+                    sub_output += '</p>'
+            html_params['entries_feedback'] = sub_output
+
+        else:
+            # Get submitted answer, raising an exception if it does not exist
+            a_sub = data['submitted_answers'].get(name, None)
+            if a_sub is None:
+                raise Exception('submitted answer is None')
+            # If answer is in a format generated by pl.to_json, convert it back to a standard type (otherwise, do nothing)
+            a_sub = pl.from_json(a_sub)
+            # Wrap answer in an ndarray (if it's already one, this does nothing)
+            a_sub = np.array(a_sub)
+            # Format answer as a string
+            if format_type == 'python':
+                html_params['a_sub'] = pl.string_from_2darray(a_sub, language='python', digits=12, presentation_type='g')
+            else:
+                html_params['a_sub'] = '$' + pl.latex_array_from_numpy_array(a_sub, presentation_type='g', digits=12) + '$'
+
+        partial_score = data['partial_scores'].get(name, {'score': None})
+        score = partial_score.get('score', None)
+        if score is not None:
+            try:
+                score = float(score)
+                if score >= 1:
+                    html_params['correct'] = True
+                elif score > 0:
+                    html_params['partial'] = math.floor(score * 100)
+                else:
+                    html_params['incorrect'] = True
+            except Exception:
+                raise ValueError('invalid score' + score)
+
+        if allow_feedback:
```

Comment:
When `allow_feedback` is `true`, the code here adds a card to the HTML titled "Feedback" that has a list of the incorrect elements (e.g. `(1, 2) (3, 1)`).

As an alternative, what about displaying the submitted answer in table format (with the same code that was used to generate the array of inputs in the question panel) and putting a red "x" next to each incorrect entry?

This would be similar to multiple choice and checkbox elements, where the "x" comes from this code:
`'&nbsp;<span class="badge badge-danger"><i class="fa fa-times" aria-hidden="true"></i></span>'`

It seems like this would be easier for students to interpret, and wouldn't require them to know that `(1, 2)` (for example) meant "first row and second column."